app/controllers/index.rb

```diff
@@ -0,0 +1,66 @@
+get '/' do
+ redirect '/suggestions'
+end
+
+get '/suggestions' do
+  @suggestions = Suggestion.all
+  erb :'suggestions/index'
+end
+
+get '/suggestions/new' do
+  # @user = User.find_by(params[:id])
+  # if @user.valid?
+  #   redirect '/'
+  # else
+  #   status 422
+  #   @errors = user.errors.full_messages
+  #   erb :'suggestions/new'
+  # end
+
+  if !logged_in?
+    redirect '/suggestions'
+  else
+    erb :'suggestions/new'
+  end
+end
+
+post '/suggestions' do
+  # binding.pry
+  if logged_in?
+    @suggestion = Suggestion.create(params[:suggestion])
+    redirect '/'
+  else
+    erb :'sugestions/new'
+  end
+end
+
+get '/suggestions/:id' do
+  # binding.pry
+  @user = User.find(params[:id])
+  @suggestion = Suggestion.find(params[:id])
```

Comment:
Watch out here: `params[:id]` is the id of the _suggestion_, but you are also using it to lookup the _user_. Get the user from the session.